### ISSUE TITLE
.bash_profile content deleted

### ISSUE THREAD

**Issue by jandubois** [rancher-sandbox/rancher-desktop#6281]
I've experienced this before on Dec 18, but it has since happened 3 more times this Friday (5-Jan-2024): After a factory-reset, and restarting Rancher Desktop with automatic `PATH` management enabled, my `~/.bash_profile` was replaced with an empty file that just had the Rancher Desktop boilerplate:

```console
$ cat ~/.bash_profile
### MANAGED BY RANCHER DESKTOP START (DO NOT EDIT)
export PATH="/Users/jan/.rd/bin:$PATH"
### MANAGED BY RANCHER DESKTOP END (DO NOT EDIT)
```

Once I perform another factory-reset, the file gets deleted completely because it is now empty.

I use symlinks into Dropbox for `~/.bash_profile` (and `~/.bashrc`):

```console
$ ls -ld ~/.bash*
-rw-------   1 jan  staff   13706  7 Jan 14:24 /Users/jan/.bash_history
lrwxr-xr-x   1 jan  staff      35  7 Jan 14:23 /Users/jan/.bash_profile -> /Users/jan/Dropbox/dot/bash_profile
drwx------  14 jan  staff     448 16 Sep  2020 /Users/jan/.bash_sessions
lrwxr-xr-x   1 jan  staff      35  7 Jan 14:23 /Users/jan/.bashrc -> /Users/jan/Dropbox/dot/bash_profile
```

It is the symlink that gets deleted on the second factory-rest.

Since the older version is still on Dropbox, it is simple to restore. Here are some screenshots from when it happened:

![CleanShot 2023-12-18 at 17 06 23@2x](https://github.com/rancher-sandbox/rancher-desktop/assets/78947/3c1fc58f-7378-4e5e-9a13-fe699dfad210)

![CleanShot 2024-01-05 at 10 50 23@2x](https://github.com/rancher-sandbox/rancher-desktop/assets/78947/bf267e85-4ce4-45db-9419-caad1889a3bf)

![CleanShot 2024-01-05 at 14 12 29@2x](https://github.com/rancher-sandbox/rancher-desktop/assets/78947/16930b04-9b9e-4cae-8732-b5fddce3cf47)

![CleanShot 2024-01-05 at 21 27 28@2x](https://github.com/rancher-sandbox/rancher-desktop/assets/78947/b9587147-1f6b-4e7b-81a7-caaaef1ffcb4)

The very last screenshot is from an instance that used manual path management, so the factory-reset didn't remove the path management logic because it was never added to the file. Still on a restart the file was overwritten.

I sometimes didn't discover that `.bash_profile` was destroyed until later (when I had to create a new terminal window), so I'm not sure if this is really true, but I believe this always happened when I initiated the factory-reset from the GUI, and restarted the app (via a hotkey) immediately once the window was closed (this was part of QASE testing, where many test cases ask for a factory-reset followed by a restart).

I've not been able to reliably reproduce the problem, even though it happened 3 times this Friday.

### Open questions

1. Is this related to having `.bash_profile` stored as a symlink?
2. Is this related to having the symlink point to Dropbox?
3. Is this related to having both `.bashrc` and `.bash_profile` symlink to the same file?
4. Does it matter that each time the app was restarted immediately after the facory reset?

### Candidate?

After looking through the sources, the only way I can see this happening is if `fs.promises.stat('~/.bash_profile')` returned `ENOENT` (or if `os.homedir()` did return an incorrect value):

https://github.com/rancher-sandbox/rancher-desktop/blob/459e1a7a3c98aeaf80e884231fb1b63448270bc1/pkg/rancher-desktop/integrations/pathManagerImpl.ts#L73-L78

But the file exists, and I don't see how `HOME` could not be set correctly, so this still doesn't make sense.

**Comment by gaktive** [comment]
@mook-as to help here to see what could be going on.

**Comment by jandubois** [comment]
And it happened again:

<img width="1091" alt="CleanShot 2024-01-08 at 14 39 52@2x" src="https://github.com/rancher-sandbox/rancher-desktop/assets/78947/408f8f04-6f44-41fb-9def-e3a1bd021415">

**Comment by jandubois** [comment]
I just noticed that at the Friday, January 5th timestamp at 10:40am my `~./zshrc` file also got truncated (it too is a symlink into Dropbox):

<img width="947" alt="CleanShot 2024-01-08 at 17 09 33@2x" src="https://github.com/rancher-sandbox/rancher-desktop/assets/78947/a138723f-fcf2-4055-9657-c5965cb38214">

**Comment by nwwells** [comment]
This happened to me on my work computer where I had not backed up my dot files yet. I'm a ZSH user, so in my case my `.zshrc` was wiped.

**Comment by jandubois** [comment]
@nwwells Thanks for letting us know! Is there anything unusual about your `~/.zshrc` in any way? Is it a symlink, like mine? If yes, does it point to Dropbox/iCloud/Google Drive/OneDrive or similar?

**Comment by nwwells** [comment]
No it was just a normal .zshrc file!

Nathan Wells

**Comment by DanielSchranc** [comment]
Hello...this has happened to me too. I installed rancher desktop with automatic PATH enabled and then it has overwritten everything I had in my bash_profile with the rancher desktop PATH

**Comment by pplanel** [comment]
I just lost my .*rc to this app. Really disappointing...

**Comment by hopsor** [comment]
I experienced the same losing my .zshrc :-/

**Comment by Cutieguwu** [comment]
Wiped my .bashrc and .bashprofile, had to restore from backup.

I don't use .zshrc (at least I don't think I do), but it didn't wipe that for me and actually appended properly.

**Comment by jandubois** [comment]
This issue is a regression from Rancher Desktop 1.11. It is a race condition that happens after either a fresh install or a factory reset, when the first-run dialog is closed again very quickly after it is displayed.

The problem has been fixed in #6352 and will be released in a 1.12.2 patch release in the next day or two, once it has passed further testing.

Sorry for not catching this regression in time!

**Comment by peimansh** [comment]
It wiped my ~/.bash_profile file in 1.14.2 on Mac ..
Wasn't this supposed to be fixed in 1.12.2 ?

my ~/.bash_profile was just a normal profile file.. no symlinks or etc ..
